pkg/machine/config.go

```diff
@@ -455,6 +457,38 @@ func (dl Download) NewFcosDownloader(imageStream FCOSStream) (DistributionDownlo
 	return fcd, nil
 }
 
+// HostHasMultipleMachines walks through the config directory to find json
+// files. It tries to unmarshal the json file into the given machine struct. If
+// unmarshalling is successful, it means the json file is a valid machine config.
+// If there are more than one of these, it is a valid assumption that there are
+// more than one machines created on the system.
+func HostHasMultipleMachines(m interface{}, confDir string) bool {
+	machineCount := 0
+
+	if err := filepath.WalkDir(confDir, func(path string, d fs.DirEntry, err error) error {
+		if strings.HasSuffix(d.Name(), ".json") {
+			b, err := os.ReadFile(filepath.Join(confDir, d.Name()))
+			if err != nil {
+				return err
+			}
+
+			if err = json.Unmarshal(b, m); err == nil {
+				// we have another machine configuration
+				machineCount++
+			}
+
+			if machineCount > 1 {
+				return nil
+			}
+		}
+		return nil
+	}); err != nil {
+		return false
+	}
```

Comment:
Doesn't this mean `HostHasMultipleMachines` always returns false, so long as the filepath walk succeeds? As both machineCount > 1 returns nil, and the fallthrough also returns nil?

Reply:
Yes, you're right. The return statement should have been `machineCount > 1` instead of `machineCount < 2`. I was returning as if I was checking if it was the only machine, not if there was more than one. Thanks for that catch!